### ISSUE TITLE
[clang-tidy] false-negative for `cppcoreguidelines-prefer-member-initialization` when initializing in the ctor body with temporary expressions/exprs with cleanup

### ISSUE THREAD

**Issue by 5chmidti** [llvm/llvm-project#81613]
Given
```c++
struct StructWithCtor {
  StructWithCtor(int x, int y) {}
};
struct AssigningFromTemporary {
    AssigningFromTemporary() : M{1, 2} { M = StructWithCtor{1, 2}; }
    StructWithCtor M;
};
StructWithCtor createTemporary();
struct AssigningFromTemporaryFromFunction {
    AssigningFromTemporaryFromFunction() : M{1, 2} { M = createTemporary(); }
    StructWithCtor M;
};
```
No diagnostics are emitted (https://godbolt.org/z/6GPYafvbd).

**Comment by PiotrZSL** [comment]
Problem with these is that if structure got custom assign operator, behavior could be different.

**Comment by 5chmidti** [comment]
Then fields where the type has non-trivial copy or move assignment (depending on what is used), can be skipped.
But then calling a constructor twice could also change the behavior. So I guess this is not actually a false-negative and not an issue.
